timemachine/fe/rbfe.py

```diff
@@ -298,27 +349,24 @@ def setup_optimized_initial_state(
     InitialState
         Optimized at the specified lambda value
     """
-
-    # NOTE: The current approach for generating optimized conformations in `optimize_coordinates` creates a
-    # discontinuity at lambda=0.5. Ensure that we pick a pre-optimized state on the same side of 0.5 as `lamb`
-    states_subset = [s for s in optimized_initial_states if (s.lamb <= 0.5) == (lamb <= 0.5)]
-    nearest_optimized = min(states_subset, key=lambda s: abs(lamb - s.lamb))
+    nearest_optimized_idx = get_nearest_state_idx(lamb, optimized_initial_states)
```

Comment:
nit: any reason for this not to return the `InitialState` itself rather than its index?

Reply:
Because I need to retrieve the associated trajectory to update the state with the coords, box and velocities.

Reply:
I might be missing something, but I don't see any other usage of `nearest_optimized_idx` in the diff?

Reply:
It is used here https://github.com/proteneer/timemachine/blob/af49751774bb234b8f5a16d9d0fa24bb44aa55a4/timemachine/fe/rbfe.py#L840